scheduling-api-graphql/scheduling-api-graphql-fetchers/src/main/java/org/ow2/proactive/scheduling/api/graphql/fetchers/HSQLDBInterceptor.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.scheduling.api.graphql.fetchers;
+
+import org.hibernate.EmptyInterceptor;
+
+
+public class HSQLDBInterceptor extends EmptyInterceptor {
+
+    @Override
+    public String onPrepareStatement(String sql) {
+        if (DatabaseConnectionFetcher.IS_HSQLDB != null && DatabaseConnectionFetcher.IS_HSQLDB) {
+            if ((sql.contains("order by") || sql.contains("ORDER BY")) &&
+                (sql.contains("LIMIT") || sql.contains("limit"))) {
+                if (sql.endsWith(";")) {
+                    return sql.substring(0, sql.length() - 1) + " USING INDEX;";
+                } else {
+                    return sql + " USING INDEX";
+                }
+            }
+            return sql;
+        } else {
+            return sql;
+        }
```

Comment:
May be you can replace this code
     return sql;
} else {
      return sql;
}
  by simply  return sql;